boards/arm/seeeduino_xiao/board.cmake

```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2020 Google LLC.
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(bossac "--offset=0x2000")
```

Comment:
Reminder that this needs to be removed if https://github.com/zephyrproject-rtos/zephyr/pull/26510 is merged before this PR (and vice versa).

Reply:
Ack.  I have a few SAMD related PRs in flight at the moment and I'll update them as they land.